src/Altinn.Platform/Altinn.Platform.Events/Events/Repository/EventsRepository.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Altinn.Platform.Events.Configuration;
+using Altinn.Platform.Events.Models;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Altinn.Platform.Events.Repository
+{
+    /// <summary>
+    /// Handles events repository. Notice that the all methods should modify the Subject attribute of the
+    /// CloudEvent, since cosmosDb fails if Subject contains slashes '/'.
+    /// </summary>
+    public class EventsRepository : IEventsRepository
+    {
+        private readonly ILogger logger;
+        private readonly Uri databaseUri;
+        private readonly Uri collectionUri;
+        private readonly string collectionId = "events";
+        private readonly string partitionKey = "/subject";
+        private readonly string triggerId = "trgUpdateItemTimestamp";
+        private readonly string databaseId;
+        private static DocumentClient client;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EventsRepository"/> class.
+        /// </summary>
+        /// <param name="cosmosettings">the configuration settings for cosmos database</param>
+        /// <param name="logger">dependency injection of logger</param>
+        public EventsRepository(IOptions<AzureCosmosSettings> cosmosettings, ILogger<EventsRepository> logger)
+        {
+            this.logger = logger;
+
+            var database = new CosmosDatabaseHandler(cosmosettings.Value);
+
+            client = database.CreateDatabaseAndCollection(collectionId);
+            collectionUri = database.CollectionUri;
+            databaseUri = database.DatabaseUri;
+            databaseId = database.DatabaseName;
+
+            DocumentCollection documentCollection = database.CreateDocumentCollection(collectionId, partitionKey);
+
+            client.CreateDocumentCollectionIfNotExistsAsync(
+                databaseUri,
+                documentCollection).GetAwaiter().GetResult();
+            InsertTrigger();
+            client.OpenAsync();
+        }
+
+        /// <inheritdoc/>
+        public async Task<string> Create(CloudEvent item)
+        {
+            ResourceResponse<Document> createDocumentResponse = await client.CreateDocumentAsync(
+                collectionUri, 
+                item, 
+                new RequestOptions { PreTriggerInclude = new List<string> { "trgUpdateItemTimestamp" } });
```

Comment:
Might be better to use triggerId here instead of the name of the trigger, so we only need to change the value one place if the trigger is to be changed.

Reply:
Aaaaah! Forgot to change when i created triggerId! Thanks :)